Runbooks/startstop/1.0/AutoStop_CreateAlert_Child.ps1

```diff
@@ -0,0 +1,321 @@
+
+        Write-Output "Processing VM ($($VMObject.Name))"
+
+        Write-Output "Current VM state is ($($VMState))"
+
+
+        $actionWebhookArm = New-AzAlertRuleWebhook -ServiceUri $webhookARMUri
+
+        if ($AlertAction -eq 'Disable') {
+            $ExVMAlerts = Get-AzMetricAlertRuleV2 -ResourceGroupName $VMObject.ResourceGroupName -ErrorAction SilentlyContinue
+
+            if ($ExVMAlerts -ne $null) {
+                Write-Output 'Checking for any previous alert(s)...'
+                #Alerts exists so disable alert
+                foreach ($Alert in $ExVMAlerts) {
+                    if ($Alert.Name.ToLower().Contains($($VMObject.Name.ToLower().Trim()))) {
+                        Write-Output "Previous alert ($($Alert.Name)) found and disabling now..."
+
+                        Get-AzMetricAlertRuleV2 -ResourceGroupName $ResourceGroupName -Name $Alert.Name | Add-AzMetricAlertRuleV2 -DisableRule
+
+                        Write-Output "Alert ($($Alert.Name)) Disabled for VM $($VMObject.Name)"
+                    }
+                }
+            }
+        } elseif ($AlertAction -eq 'Create') {
+            Write-Output 'Creating alerts...'
+            $VMAlerts = Get-AzMetricAlertRuleV2 -ResourceGroupName $ResourceGroupName -ErrorAction SilentlyContinue
+
+            #Check if alerts exists and take action
+            if ($VMAlerts -ne $null) {
+                Write-Output 'Checking for any previous alert(s)...'
+                #Alerts exists so delete and re-create the new alert
+                foreach ($Alert in $VMAlerts) {
+
+                    if ($Alert.Name.ToLower().Contains($($VMObject.Name.ToLower().Trim()))) {
+                        Write-Output "Previous alert ($($Alert.Name)) found and deleting now..."
+                        #Remove the old alert
+                        Remove-AzMetricAlertRuleV2 -Name $Alert.Name -ResourceGroupName $ResourceGroupName
+
+                        #Wait for few seconds to make sure it processed
+                        Do {
+                            #Start-Sleep 10
+                            $GetAlert = Get-AzMetricAlertRuleV2 -ResourceGroupName $ResourceGroupName -Name $Alert.Name -ErrorAction SilentlyContinue
+                        }
+                        while ($GetAlert -ne $null)
+
+                        Write-Output 'Generating a new alert with unique name...'
+                        #Now generate new unique alert name
+                        $NewAlertName = Generate-AlertName -OldAlertName $Alert.Name -VMName $VMObject.Name
+                    }
+                }
+
+            }
+
+            #Alert does not exist, so create new alert
+            Write-Output $NewAlertName
+
+            Write-Output 'Adding a new alert to the VM...'
+
+            #Creating ARM alert is multi-step process.
+
+            #1. Check for existing action group first and reuse if not else Create the Action group receiver and Action group.
```

Comment:
WHAT

Reply:
Microsoft created this script. We can optimize and fix them in a different PR.